lib/TickingFixedClock.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Lendable\Clock;
+
+/**
+ * Simple in-memory ticking fixed clock.
+ *
+ * The original fixed value will advance forward as normal to simulate time passing from
+ * the fixed time point.
+ */
+final class TickingFixedClock implements MutableClock
+{
+    private \DateTimeImmutable $now;
+
+    private function __construct(\DateTimeInterface $now, private \DateTimeImmutable $tickingFrom)
+    {
+        $this->now = \DateTimeImmutable::createFromInterface($now);
+    }
+
+    public static function tickingFromCurrentTime(\DateTimeImmutable $now): self
+    {
+        return self::tickingFromTime($now, new \DateTimeImmutable('now'));
+    }
+
+    /**
+     * Constructs with a specific point in time the clock was originally began ticking from.
+     *
+     * This allows for testing where you are persisting state of a clock across in-memory boundaries
+     * and need to re-hydrate into that state, rather than begin ticking time from the current time again.
+     */
+    public static function tickingFromTime(\DateTimeImmutable $now, \DateTimeImmutable $tickingFrom): self
+    {
+        return new self($now, $tickingFrom);
+    }
+
+    public function now(): \DateTimeImmutable
+    {
+        return $this->now->add($this->calculateOffset());
+    }
+
+    public function nowMutable(): \DateTime
+    {
+        return \DateTime::createFromImmutable($this->now());
+    }
+
+    public function changeTimeTo(\DateTimeInterface $time): void
+    {
+        $this->now = \DateTimeImmutable::createFromInterface($time);
+        $this->tickingFrom = new \DateTimeImmutable('now');
```

Comment:
is this valid behaviour in all cases..? Should you override this value when instance was initially constructed through `tickingFromTime`?

Reply:
Yeah you need to for it to behave as expected. The manual construction with a value is just for persistence support so you can continue with the correct offset when re-constructing. 

When you change the time, you're resetting the point you are fixing from, you'd not expect:

```php
$clock = TickingFixedClock::tickFromCurrentTime(...);
$clock->now() // <1ms difference from fixed value.

// persist clock state.
// 5 seconds passes.

$clock->changeTimeTo(/* some other time point */);
$clock->now() // time is off by >5 secs to what it was just changed to.
```

Essentially there's no use case apart from serialized state being required for calling the `tickingFromTime(...)`

Reply:
What if I do not care about the assumed purpose of that method (persistence) and would expect it to be off by 5 seconds when I time travel using that method? Why else would I time travel?

Reply:
Can document it internal or rework to make it not part of a public API?

I'd argue you only time travel when you want the clock to be set to a specific time and then tick from there again, as a physical clock would when the time is changed.

Example would be sending a bunch of commands into a system on a certain date, then time travelling forward to run a scheduler for follow up commands that were created as a result of the original command handling.

Reply:
Can also drop it from this PR and re-introduce as part of persistence/serialisation support PR.